tests/utils/mock-server-client/Client.h

```diff
@@ -0,0 +1,120 @@
+const auto kExpectationPath = "/mockserver/expectation";
+const auto kStatusPath = "/mockserver/status";
+const auto kResetPath = "/mockserver/reset";
+const auto kTimeout = std::chrono::seconds{10};
+
+class Client {
+ public:
+  explicit Client(const olp::client::OlpClientSettings& settings) {
+    http_client_ = olp::client::OlpClientFactory::Create(settings);
+    http_client_->SetBaseUrl(kBaseUrl);
+  }
+
+  void MockResponse(const std::string& method_matcher,
+                    const std::string& path_matcher,
+                    const std::string& response_body) {
+    auto expectation = Expectation{};
+    expectation.request.path = path_matcher;
+    expectation.request.method = method_matcher;
+
+    boost::optional<Expectation::ResponseAction> action =
+        Expectation::ResponseAction{};
+    action->body = response_body;
+    expectation.action = action;
+
+    CreateExpectation(expectation);
+  }
+
+  std::vector<int32_t> Ports() const {
+    auto response = http_client_->CallApi(
+        kStatusPath, "PUT", std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(), nullptr, "",
+        olp::client::CancellationContext{});
+
+    if (response.status != olp::http::HttpStatusCode::OK) {
+      return {};
+    }
+
+    auto response_body = std::string{};
+    response.GetResponse(response_body);
+    const auto status = olp::parser::parse<Status>(response_body);
+
+    return status.ports;
+  }
+
+  void Reset() {
+    auto response = http_client_->CallApi(
+        kResetPath, "PUT", std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(), nullptr, "",
+        olp::client::CancellationContext{});
+
+    return;
+  }
+
+ private:
+  void CreateExpectation(const Expectation& expectation) {
+    const auto data = serialize(expectation);
+    std::cout << data << std::endl;
+    const std::shared_ptr<std::vector<unsigned char>> request_body =
+        std::make_shared<std::vector<unsigned char>>(data.begin(), data.end());
+
+    auto response = http_client_->CallApi(
+        kExpectationPath, "PUT", std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(),
+        std::multimap<std::string, std::string>(), request_body, "",
+        olp::client::CancellationContext{});
+
+    return;
+  }
+
+ private:
+  std::shared_ptr<olp::client::OlpClient> http_client_;
+};
+
+static std::unique_ptr<mockserver::Client> CreateDefaultMockServerClient(){
+  auto settings = olp::client::OlpClientSettings{};
+  settings.network_request_handler = olp::client::OlpClientSettingsFactory::
+      CreateDefaultNetworkRequestHandler();
+  settings.task_scheduler =
+      olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler();
+  return std::make_unique<mockserver::Client>(settings);
+}
```

Comment:
I would expect we share the Network layer with the clients. Maybe take the OlpClientSettings as input?